__tests__/utils/permission.test.js

```diff
@@ -0,0 +1,102 @@
+    const checkPermission = await checkCameraPermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual('');
+  });
+});
+
+describe('requestExternalStoragePermission shpuld run correctly', () => {
+  beforeAll(() => {
+    jest.mock('react-native/Libraries/Utilities/Platform', () => ({
+      OS: 'android', // or 'ios'
+      select: () => null
+    }));
+  });
+  it('requestExternalStoragePermission granted should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('granted');
+    const checkPermission = await requestExternalStoragePermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionGranted);
+  });
+  it('requestExternalStoragePermission unavailable should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('unavailable');
+    const checkPermission = await requestExternalStoragePermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionUnavailable);
+  });
+  it('requestExternalStoragePermission denied should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('denied');
+    const checkPermission = await requestExternalStoragePermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionGranted);
+  });
+  it('requestExternalStoragePermission default should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('default');
+    const checkPermission = await requestExternalStoragePermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual('');
+  });
+  it('requestExternalStoragePermission denied should run correctly', async () => {
+    jest.spyOn(mock, 'check').mockResolvedValue('denied');
+    jest.spyOn(mock, 'request').mockResolvedValue('denied');
+    const checkPermission = await requestExternalStoragePermission();
+    expect(mock.check).toHaveBeenCalled();
+    expect(mock.request).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionDenied);
+  });
+});
+
+describe('request camera permission should run correctly', () => {
+  it('request denied should run correctly', async () => {
+    jest.spyOn(mock, 'request').mockResolvedValue('denied');
+    const checkPermission = await requestCameraPermission();
+    expect(mock.request).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionDenied);
+  });
+
+  it('request granted should run correctly', async () => {
+    jest.spyOn(mock, 'request').mockResolvedValue('granted');
+    const checkPermission = await requestCameraPermission();
+    expect(mock.request).toHaveBeenCalled();
+    expect(checkPermission.message).toEqual('');
+  });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The tests are well written and cover all the possible outcomes of the permission checks. However, there seems to be a typo in line 47: "shpuld" should be "should". Also, in the test cases for `requestExternalStoragePermission`, it seems like you're using `StringConstant.cameraPermissionGranted` and `StringConstant.cameraPermissionUnavailable` instead of constants specific to external storage permissions. If these constants are not available, consider adding them for better clarity.

```diff
-47: describe('requestExternalStoragePermission shpuld run correctly', () => {
+47: describe('requestExternalStoragePermission should run correctly', () => {

...

-58:     expect(checkPermission.message).toEqual(StringConstant.cameraPermissionGranted);
+58:     expect(checkPermission.message).toEqual(StringConstant.externalStoragePermissionGranted);

...

-64:     expect(checkPermission.message).toEqual(StringConstant.cameraPermissionUnavailable);
+64:     expect(checkPermission.message).toEqual(StringConstant.externalStoragePermissionUnavailable);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->